stats/system_tag.go

```diff
@@ -0,0 +1,134 @@
+package stats
+
+import (
+	"bytes"
+	"encoding/json"
+	"strings"
+)
+
+// SystemTagSet is a bitmask that is used to keep track
+// which system tags should be included with which metrics.
+//go:generate enumer -type=SystemTagSet -transform=snake -trimprefix=Tag -output system_tag_set_gen.go
+type SystemTagSet uint32
+
+// SystemTagMap is a string to bool map (for lookup efficiency) that is used to keep track
+// which system tags should be included with with metrics.
+type SystemTagMap map[string]bool
+
+// DefaultSystemTagList includes all of the system tags emitted with metrics by default.
+// Other tags that are not enabled by default include: iter, vu, ocsp_status, ip
+//nolint:gochecknoglobals
+var DefaultSystemTagList = []string{
+	TagProto.String(),
+	TagSubProto.String(),
+	TagStatus.String(),
+	TagMethod.String(),
+	TagURL.String(),
+	TagName.String(),
+	TagGroup.String(),
+	TagCheck.String(),
+	TagCheck.String(),
+	TagError.String(),
+	TagErrorCode.String(),
+	TagTLSVersion.String(),
+}
+
+//nolint: golint
+const (
+	// Default system tags includes all of the system tags emitted with metrics by default.
+	TagProto SystemTagSet = 1 << iota
+	TagSubProto
```

Comment:
Hmm, `enumer` only have `snake` and `kebab`, any suggestion?

Reply:
I agree, it's going to be slightly inconsistent and annoying to have `TagSubproto` from the standpoint of k6 code, but we don't have a choice, since the `string` tag names **must** be backwards compatible. That's the user-visible part that we absolutely need to keep the same, since the whole purpose of this PR is to keep the same user-facing behavior, but have better performance and less `string`-y code on the k6 side.

Also, I'm not insisting that we use `enumer` here, it was only a suggestion in the original issue. There's a benefit to using only a single tool in the k6 codebase to generate these types, but if there's another tool that has more flexible generation of Go code and there's a good reason to switch, we can start here and maybe switch the other types at a later time. Not sure a single capital letter is worth it in this case, but there might be other benefits.

Reply:
@na-- maybe fork enumer and tailor it for our specific case. I don't know (and think don't have) any existing tool that satisfy this specific case for us.

Reply:
I'm very much against forking such a tool for such a trivial reason :smile: If you're very bothered by the improper capitalization of a single letter, add a `TODO` in our code, make a PR to the original enumer project that adds a good way to fix it, and when that's merged - come back and fix the `TODO` :wink: 

It's also worth pointing out that enumer is just a time saving tool, there's nothing stopping us from actually manually doing something similar to its generated code, but again, in this case, I don't think it's worth it...